### ISSUE TITLE
pytest: add test decorator for fiddling the return values of glbl_cfg

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3392]
Cylc has a global configuration, in functional tests we often change configuration values by creating a temporary file and changing the `CYLC_CONF_PATH` environment variable.

This seems a little heavy-handed so in the unit tests we sometimes mock the output of `cylc.flow.cfgspec.globalcfg.GlobalConfig.get_host_item`, objects e.g:

```python
    @patch('cylc.flow.pathutil.glbl_cfg')
    def test_get_remote_suite_run_dirs(self, mocked_glbl_cfg):
        """Usage of get_remote_suite_run_*dir."""
        mocked = MagicMock()
        mocked_glbl_cfg.return_value = mocked
        mocked.get_host_item.return_value = '/home/sweet/cylc-run'
        # ...
```

It would be nice if we could centralise this approach to:

* Make it easier to update tests when the global config interface changes.
* Make tests safer by targeting the specific settings we want to override.
* Facilitate unit testing.

Ideally we could write a decorator something like this:

```python
    @glbl_cfg('cylc.flow.pathutil', {('cylc', 'suite run dir'): '/home/sweet/cylc-run'})
    def test_get_remote_suite_run_dirs(self, mocked_glbl_cfg):
        """Usage of get_remote_suite_run_*dir."""
        # ...
```

Note that in the codebase modules import the `glbl_cfg` function which is for lazy loading of the global config object.

**Pull requests welcome!**

**Comment by oliver-sanders** [comment]
@wxtim have you closed this?

**Comment by oliver-sanders** [comment]
Closed by #3489
